docs/adr/0009_versioning_branching_releasing.md

```diff
@@ -0,0 +1,465 @@
+
+#### Release Workflow for Sub-Components
+
+Sub-components follow a two-phase release process:
+
+##### Phase 1: Release Candidate Creation
+
+* Create and test release candidates from release branches
+* Generate RC tags for validation: `<component>/v<major>.<minor>.<patch>-rc.N`
+* Extended testing and validation period before final release
+
+##### Phase 2: Final Release Promotion
+
+* Promote validated RC to final release
+* Create final Git tags: `<component>/v<major>.<minor>.<patch>`
+* Automatically trigger OCM root component release candidate evaluation
+* Publish final artifacts and release notes
+
+*Detailed technical workflow steps are described in the [CI/CD Workflows section](#workflow-sub-component-release).*
+
+#### Hotfix/Backport Procedure
+
+**Process**:
+
+1. Implement fix on the appropriate `releases/<component>/vX.Y` branch
+2. Open PR targeting the release branch
+3. After merge, backport change to `main` branch (cherry-pick or separate PR) if applicable
+
+**Example**: Hotfix for CLI v0.30.2 goes to `releases/cli/v0.30` → `cli/v0.30.2`
+
+#### Sub-Component Release Types
+
+All sub-component releases follow a streamlined process with RC workflow only required for initial minor version releases:
+
+* **Minor Version Releases**: Create new release branch → RC workflow → final release (e.g., cli/v0.30.0)
+* **Patch Releases**: Use existing release branch → direct final release (e.g., cli/v0.30.1, cli/v0.30.2)
+* **RC Requirement**: Only the initial minor version (.0) requires RC workflow; subsequent patches are released directly from the release branch
+* **Impact on `ocm` root component**: final release tag triggers new OCM RC creation with incremented number
+
+### OCM Root Component Strategy
+
+The `ocm` root component utilizes an **OCM Component Constructor YAML** located in `/ocm/component-constructor.yaml` as the authoritative version matrix. This constructor:
+
+* References specific versions of all sub-components (cli, controller, etc.)
+* Acts as the definitive snapshot for reproducible OCM releases
+* Is automatically updated via bot PRs when sub-component releases trigger OCM releases
+
+#### Template for `/ocm/component-constructor.yaml`
+
+```yaml
+components:
+  - name: ocm.software/ocm
+    version: ${OCM_VERSION}  # will be resolved at release time
+    provider: ocm.software
+    
+    componentReferences:
+      - name: cli
+        componentName: ocm.software/cli
+        version: ${OCM_CLI_VERSION}  # will be resolved at release time
+
+      - name: controller
+        componentName: ocm.software/kubernetes/controller
+        version: ${OCM_CONTROLLER_VERSION}  # will be resolved at release time
+```
+
+### OCM Root Component Release Process
+
+**Key Principle**: Any sub-component release of type **patch** and **minor** triggers an OCM **minor** version bump, as each represents a new combination of tested components. A **major** release of a sub-component triggers an OCM **major** version bump.
```

Comment:
That kinda seems excessive on a patch version? 🤔 Even if it is a patch version we release a new minor version? That means that every release of the root ocm component is potentially a breaking change. What's the point of a patch release for a component then?

Reply:
See my other comment. @jakobmoellerdev asked to keep it the setup for the root component simple and always increase the minor version